pkg/storage/sqlcommon/sqlcommon.go

```diff
@@ -728,3 +765,31 @@ func AddFromUlid(sb sq.SelectBuilder, fromUlid string, sortDescending bool) sq.S
 	}
 	return sb.Where(sq.Gt{"ulid": fromUlid})
 }
+
+// HandleSQLError processes an SQL error and converts it into a more
+// specific error type based on the nature of the SQL error.
+func HandleSQLError(err error, args ...interface{}) error {
+	if errors.Is(err, sql.ErrNoRows) {
+		return storage.ErrNotFound
+	}
+
+	if me, ok := err.(*mysql.MySQLError); ok && me.Number == 1062 {
```

Comment:
should we combine the logic with line 785-792?

Reply:
You could, when you want to handle another one it will get messy. Better for readability to keep them separated.